### PR TITLE
[utility] Filter FDB entries

### PR DESCRIPTION
**Pull request by tahmed-dev**
**- What I did**
FDB table can get large due to VM creation/deletion which cause
fast reboot to slow down. This utility fitlers FDB entries based on
current MAC entries in the ARP table.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

**- How I did it**
code change

**- How to verify it**
run unit test
sudo pytest sonic-utilities-tests/filter_fdb_entries_test.py

**- Previous command output (if the output of a command-line utility has changed)**
N/A
**- New command output (if the output of a command-line utility has changed)**
N/A

### CODE REVIEW

File: scripts/filter_fdb_entries.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-f', '--fdb', type=str, default='/tmp/fdb.json', help='fdb file name')
+    parser.add_argument('-a', '--arp', type=str, default='/tmp/arp.json', help='arp file name')
+    parser.add_argument('-b', '--backup_file', type=bool, default=True, help='Back up old fdb entries file')
+    args = parser.parse_args()
+
+    fdb_filename = args.fdb
+    file_exits_or_raise(fdb_filename)
```
Comment (yxieca) [inline scripts/filter_fdb_entries.py:95]: Let's return with a log message. I don't think adding a failure case in fast reboot is good.

File: scripts/filter_fdb_entries.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    arp_filename = args.arp
+    file_exits_or_raise(arp_filename)
```
Comment (yxieca) [inline scripts/filter_fdb_entries.py:98]: I think we should return here too.

**Comment by yxieca** [comment]
Where is the invocation of the script?

File: scripts/filter_fdb_entries.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    res = 0
+    try:
+        syslog.openlog('filter_fdb_entries')
+        res = main()
+    except KeyboardInterrupt:
+        syslog.syslog(syslog.LOG_NOTICE, "SIGINT received. Quitting")
+        res = 1
+    except Exception as e:
+        syslog.syslog(syslog.LOG_ERR, "Got an exception %s: Traceback: %s" % (str(e), traceback.format_exc()))
+        res = 2
```
[the remark this answers is not in the document]
Comment (yxieca) [reply scripts/filter_fdb_entries.py:118]: Yes, but the return code 2 will cause the fast-reboot to abort. If we only have 1 of the 2 files. I don't know if that is legitimate quitting condition.

[the remark this answers is not in the document]
Comment (tahmed-dev) [reply scripts/filter_fdb_entries.py:118]: I update the PR and now, the code is not throwing if either or both of the files are missing. This throw could happen if the filter routine itself throws due to badly formatted file.

**Pull request merged**